internal/agent/install.go

```diff
@@ -54,25 +54,33 @@ func displayInfo(agentConfig *Config) {
 }
 
 func ManualInstall(c, sourceImgURL, device string, reboot, poweroff, strictValidations bool) error {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	configSource, err := prepareConfiguration(ctx, c)
-	if err != nil {
-		return err
+	var cfg io.Reader
+
+	// Config can be either a file in the system or an url
+	// We need to differentiate between the two
+	// If its a local file, we jsut read it and pass it to the config.Scan
+	// If its a url, we need to create a configuration with the url and let the config.Scan handle it
+	if u, err := url.Parse(c); err != nil || u.Scheme == "" {
+		file, err := os.ReadFile(c)
+		if err != nil {
+			return err
+		}
+		cfg = bytes.NewReader(file)
+	} else {
+		cfgUrl := fmt.Sprintf(`config_url: %s`, c)
```

Comment:
I still don't see why the old code wasn't working. It was constructing a file on disk with the exact same option no?